include/deal.II/base/partitioner.templates.h

```diff
@@ -402,7 +402,11 @@ namespace Utilities
 
       // clear the ghost array in case we did not yet do that in the _start
       // function
-      std::memset(ghost_array.begin(), 0, sizeof(Number)*n_ghost_indices());
+      if (ghost_array.size()>0)
+        {
+          Assert(ghost_array.begin()!=nullptr, ExcInternalError());
```

Comment:
This assumes that `.begin()` returns a pointer. But iterators could have other types.

I'd just drop the assertion -- if the size is non-zero, surely `begin()` returns something useful.

Reply:
All the `mem`* functions expect `void*` types. Therefore, I assumed that this check can't be a problem.

Reply:
Fair enough.

`ghost_array` is an `ArrayView`. We should probably give it a `.data()` member function at one point or other to make it look more like a `std::vector`. But that can wait for some other time.